CRM/Event/Form/Registration/Register.php

```diff
@@ -651,18 +651,14 @@ public static function buildAmount(&$form, $required = TRUE, $discountId = NULL)
           }
 
           $eventFeeBlockValues['amount_id_' . $fee['amount_id']] = $fee['value'];
-          $elements[] = &$form->createElement('radio', NULL, '',
-            CRM_Utils_Money::format($fee['value']) . ' ' .
-            $fee['label'],
-            $fee['amount_id'],
-            $totalAmountJs
-          );
+          $elements[$fee['amount_id']] = CRM_Utils_Money::format($fee['value']) . ' ' . $fee['label'];
+          $elementJS[$fee['amount_id']] = $totalAmountJs;
         }
       }
       $form->assign('eventFeeBlockValues', json_encode($eventFeeBlockValues));
 
       $form->_defaults['amount'] = $form->_values['event']['default_fee_id'] ?? NULL;
-      $element = &$form->addGroup($elements, 'amount', ts('Event Fee(s)'), '<br />');
+      $element = &$form->addRadio('amount', ts('Event Fee(s)'), $elements, [], '<br />', FALSE, $elementJS);
```

Comment:
Just eyeballing looks ok, but I can't figure out how to get here. Everything always gets treated as a price set even when not using price sets.